library/src/scala/tasty/Reflection.scala

```diff
@@ -2385,21 +2385,25 @@ trait Reflection { reflection =>
   // IMPLICIT SEARCH //
   /////////////////////
 
-  // TODO: this should not be top level
-  /** Find an implicit of type `T` in the current scope given by `ctx`.
-   *  Return an `ImplicitSearchResult`.
-   *
-   *  @param tpe type of the implicit parameter
-   *  @param ctx current context
-   */
-  def searchImplicit(tpe: Type): ImplicitSearchResult
+  /** Result of a given instance search */
+  type ImplicitSearch <: AnyRef
+
+  given TypeTest[ImplicitSearch, ImplicitSearchSuccess] = ImplicitSearchSuccessTypeTest
+  protected val ImplicitSearchSuccessTypeTest: TypeTest[ImplicitSearch, ImplicitSearchSuccess]
 
-  type ImplicitSearchResult <: AnyRef
+  val ImplicitSearch: ImplicitSearchModule
 
-  given TypeTest[ImplicitSearchResult, ImplicitSearchSuccess] = ImplicitSearchSuccessTypeTest
-  protected val ImplicitSearchSuccessTypeTest: TypeTest[ImplicitSearchResult, ImplicitSearchSuccess]
+  trait ImplicitSearchModule { self: ImplicitSearch.type =>
+    /** Find a given instance of type `T` in the current scope.
+     *  Return an `ImplicitSearch`.
```

Comment:
Will it find Scala2 `implicit` instances? The "current scope' needs some explanation.

Reply:
Maybe change `the current scope` to `the current scope (e.g., the call site of a macro)`?